portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,59 +20,128 @@
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.lang.Integer;
 import java.util.ArrayList;
 import java.util.List;
 import com.google.gson.Gson;
-import com.google.appengine.api.datastore.FetchOptions;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+
   //class methhods: convert to JSon and get parameter
-  private String convertToJson(ArrayList<String> commentList) {
+  private String convertToJsonStrArrayList(ArrayList<String> commentList) {
     Gson gson = new Gson();
     return gson.toJson(commentList);
   }
 
+  private String convertToJson(ArrayList<String> stats, ArrayList<Comment> commentList) {
+    String ret = "[";
+    ret += "{";
+    ret += "\"total-comments\": " + stats.get(0) + ",";
+    ret += "\"avg-rating\": " + stats.get(1) + ",";
+    ret += "\"nps\": " + stats.get(2);
+    ret += "},";
+    for (Comment c : commentList) {
+        ret += c.convertToJson();
+        ret += ",";
+    }
+    ret = ret.substring(0, ret.length() - 1);
+    ret += "]";
+    return ret;
+  }
+
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     if (value == null) {
       return defaultValue;
     }
     return value;
   }
-  // our datastructure for now.... very simple
-  private ArrayList<String> commentList = new ArrayList<String>();//Creating arraylist
+
+  private Query getQuerySort(String sort_method) {
+    if (sort_method.equals("ts_desc")) {
+        return new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    }
+    else if (sort_method.equals("ts_asc")) {
+        return new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    }
+    else if (sort_method.equals("rtg_desc")) {
+        return new Query("Comment").addSort("rating", SortDirection.DESCENDING);
+    }
+    else if (sort_method.equals("rtg_asc")) {
+        return new Query("Comment").addSort("rating", SortDirection.ASCENDING);
+    }
+    else if (sort_method.equals("author_desc")) {
+        return new Query("Comment").addSort("name", SortDirection.DESCENDING);
+    }
+    else if (sort_method.equals("author_asc")) {
+        return new Query("Comment").addSort("name", SortDirection.ASCENDING);
+    }
+    else if (sort_method.equals("comm_len_desc")) {
+        return new Query("Comment").addSort("comment_length", SortDirection.DESCENDING);
+    }
+    else if(sort_method=="comm_len_asc") {
+        return new Query("Comment").addSort("comment_length", SortDirection.ASCENDING);
+    }
+    return new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+  }
+
+  private static double round (double value, int precision) {
```

Comment:
you only use the output of this as a string anyway. I would consider using DecimalFormat, e.g.

https://stackoverflow.com/questions/153724/how-to-round-a-number-to-n-decimal-places-in-java